runtime/src/jvmMain/kotlin/dev/zacsweers/metro/internal/InstanceFactory.jvm.kt

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2014 Zac Sweers
+// SPDX-License-Identifier: Apache-2.0
+package dev.zacsweers.metro.internal
+
+@JvmInline
```

Comment:
Why not add this annotation on the expect class so you don't even need the actual classes?

Reply:
Only available in the JVM artifact isn't it?

Reply:
It's available in common but only used during the JVM compilation.

Reply:
well that makes things easier!

Reply:
Continued in #297